components/event_dispatcher/introduction.rst

```diff
@@ -268,57 +245,28 @@ and dispatch your own events. This is useful when creating third-party
 libraries and also when you want to keep different components of your own
 system flexible and decoupled.
 
-The Static ``Events`` Class
-...........................
-
-Suppose you want to create a new Event - ``store.order`` - that is dispatched
-each time an order is created inside your application. To keep things
-organized, start by creating a ``StoreEvents`` class inside your application
-that serves to define and document your event::
-
-    namespace Acme\StoreBundle;
-
-    final class StoreEvents
-    {
-        /**
-         * The store.order event is thrown each time an order is created
-         * in the system.
-         *
-         * The event listener receives an
-         * Acme\StoreBundle\Event\FilterOrderEvent instance.
-         *
-         * @var string
-         */
-        const STORE_ORDER = 'store.order';
-    }
-
-Notice that this class doesn't actually *do* anything. The purpose of the
-``StoreEvents`` class is just to be a location where information about common
-events can be centralized. Notice also that a special ``FilterOrderEvent``
-class will be passed to each listener of this event.
-
-Creating an Event Object
-........................
+Creating an Event Class
+.......................
 
-Later, when you dispatch this new event, you'll create an ``Event`` instance
-and pass it to the dispatcher. The dispatcher then passes this same instance
-to each of the listeners of the event. If you don't need to pass any
-information to your listeners, you can use the default
-``Symfony\Component\EventDispatcher\Event`` class. Most of the time, however,
-you *will* need to pass information about the event to each listener. To
-accomplish this, you'll create a new class that extends
-``Symfony\Component\EventDispatcher\Event``.
+Suppose you want to create a new event - ``store.order_placed`` - that is
+dispatched each time a customer orders a product with your application. When
+dispatching this event, you'll pass a custom event instance that has access to
+the placed order. Start by creating this custom event class and documenting
+it::
 
-In this example, each listener will need access to some pretend ``Order``
-object. Create an ``Event`` class that makes this possible::
-
-    namespace Acme\StoreBundle\Event;
+    namespace Acme\Store\Event;
 
     use Symfony\Component\EventDispatcher\Event;
-    use Acme\StoreBundle\Order;
+    use Acme\Store\Order;
 
-    class FilterOrderEvent extends Event
+    /**
+     * The store.order_placed event is dispatched each time an order is created
+     * in the system.
+     */
+    class OrderPlacedEvent extends Event
     {
+        const NAME = 'store.order_placed';
```

Comment:
Instead of creating a static events class with all event names, I propose to put the event names as a constant in each event class. This means it's easier to dispatch (no need to use 2 different classes) and everything is located in the same file.

This also introduces the better practice to have one event object correspond to only one event name.